src/lightning_app/storage/drive.py

```diff
@@ -142,6 +163,10 @@ def get(
         """
         if _is_flow_context():
             raise Exception("The flow isn't allowed to get files from a Drive.")
+        if self.protocol == "s3://":
+            raise PermissionError(
+                "S3 based drives cannot currently get files via this API. Did you mean to use `lit://` drives?"
+            )
```

Comment:
@tchaton @rlizzo Why/how would one use a drive if all these basic operations are not supported. Can't put files and can't get files, what's the point xD

Reply:
This Drive type is only for mounting S3 buckets on the Work.

The original direction was to have the ability to mount drives on Work RO. Then to also allow users to mount S3 locations to Work RO as well.

My general disagreement is with calling them "S3 drives", I think it's an antipattern and it won't stand the test of time really quickly. I have already expressed it in a platform PR.

Reply:
These are mounted drives that will be visible as a file system at the `root_folder` location of the work. You can interact with the file system normally via any means with which you'd interact with files on the local machine. (ie. using dataloaders etc)